src/sagas/signoff.ts

```diff
@@ -45,106 +45,116 @@ export function* onCollectionRecordsRequest(
+      lastEditBy,
+      lastEditDate,
+      lastEditorComment,
+      lastReviewRequestBy,
+      lastReviewRequestDate,
+      lastReviewBy,
+      lastReviewDate,
+      lastReviewerComment,
+      lastSignatureBy,
+      lastSignatureDate,
+      status,
+      editors_group,
+      reviewers_group,
+    };
+    const collectionsInfo = {
+      ...basicInfos,
+      source: sourceInfo,
+    };
+    yield put(SignoffActions.workflowInfo(collectionsInfo));
 
-  // If no preview collection / no review enabled: do not fetch changes.
-  if (!preview) {
-    // Note that we don't really have to support collections with preview disabled UI.
-    // The collections with review disabled are very likely to be manipulated by scripts anyway.
-    return;
-  }
+    // If no preview collection / no review enabled: do not fetch changes.
+    if (!preview) {
+      // Note that we don't really have to support collections with preview disabled UI.
+      // The collections with review disabled are very likely to be manipulated by scripts anyway.
+      return;
+    }
 
-  let changesOnSource = null;
-  let changesOnPreview = null;
+    let changesOnSource = null;
+    let changesOnPreview = null;
 
-  // Figure out what was changed on the source collection since review request.
-  // There can be changes on source only if status is work-in-progress.
-  if (sourceAttributes.status == "work-in-progress") {
-    changesOnSource = yield call(fetchChangesInfo, source, preview);
-    // Figure out what was changed on the preview collection since last approval.
-    // Don't bother fetching changes if current collection is signed (no pending changes).
-  } else if (status != "signed") {
-    changesOnPreview = yield call(fetchChangesInfo, source, destination);
-  } else {
-    // Status is signed. No changes to show.
-    return;
-  }
+    // Figure out what was changed on the source collection since review request.
+    // There can be changes on source only if status is work-in-progress.
+    if (sourceAttributes.status == "work-in-progress") {
+      changesOnSource = yield call(fetchChangesInfo, source, preview);
+      // Figure out what was changed on the preview collection since last approval.
+      // Don't bother fetching changes if current collection is signed (no pending changes).
+    } else if (status != "signed") {
+      changesOnPreview = yield call(fetchChangesInfo, source, destination);
+    } else {
+      // Status is signed. No changes to show.
+      return;
+    }
 
-  const collectionsInfoWithChanges = {
-    ...collectionsInfo,
-    changesOnSource,
-    changesOnPreview,
-  };
-  yield put(SignoffActions.workflowInfo(collectionsInfoWithChanges));
+    const collectionsInfoWithChanges = {
+      ...collectionsInfo,
+      changesOnSource,
+      changesOnPreview,
+    };
+    yield put(SignoffActions.workflowInfo(collectionsInfoWithChanges));
+  } catch (ex) {
+    if (ex.data?.code === 401) {
+      console.warn(ex);
```

Comment:
Can't we just do this around the call to `yield call(fetchSourceAttributes, source);` ?

Reply:
Nope, unfortunately. This was fun to troubleshoot, but there were multiple places in this saga that can end up giving us a 401 response. (ex: SignoffActions.workflowInfo can as well, and that's called from several places).

I figured one big try/catch was better than 10 small ones since the end result was all the same.